src/plugin/featureaction/ui/featuresoundactionconfig.js

```diff
@@ -0,0 +1,96 @@
+goog.provide('plugin.im.action.feature.ui.SoundConfigCtrl');
+goog.provide('plugin.im.action.feature.ui.soundConfigDirective');
+
+goog.require('os.audio.AudioManager');
+goog.require('os.object');
+goog.require('os.ui.Module');
+goog.require('os.ui.im.action.EventType');
+goog.require('os.ui.spinnerDirective');
+goog.require('os.ui.uiSwitchDirective');
+goog.require('plugin.im.action.feature.SoundAction');
+goog.require('plugin.im.action.feature.ui.ActionConfigCtrl');
+
+/**
+ * Directive to configure a feature style action.
+ * @return {angular.Directive}
+ *
+ */
+plugin.im.action.feature.ui.soundConfigDirective = function() {
+  return {
+    restrict: 'E',
+    replace: true,
+    templateUrl: os.ROOT + 'views/plugin/featureaction/featuresoundactionconfig.html',
+    controller: plugin.im.action.feature.ui.SoundConfigCtrl,
+    controllerAs: 'ctrl'
+  };
+};
+
+/**
+ * Add the directive to the module.
+ */
+os.ui.Module.directive(plugin.im.action.feature.SoundAction.CONFIG_UI,
+    [plugin.im.action.feature.ui.soundConfigDirective]);
+
+/**
+ * Controller for setting a feature sound.
+ * @param {!angular.Scope} $scope The Angular scope.
+ * @param {!angular.JQLite} $element The root DOM element.
+ * @extends {plugin.im.action.feature.ui.ActionConfigCtrl<plugin.im.action.feature.SoundAction>}
+ * @constructor
+ * @ngInject
+ */
+plugin.im.action.feature.ui.SoundConfigCtrl = function($scope, $element) {
+  plugin.im.action.feature.ui.SoundConfigCtrl.base(this, 'constructor', $scope,
+      $element);
+  $scope['sounds'] = os.audio.AudioManager.getInstance().getSounds();
```

Comment:
- When creating new properties for the template that are not already on the scope, we generally prefer putting them on the controller (so, `this['sounds'] = ....`), then reference them in the template as `ctrl.sounds`. The `controllerAs` property in the template adds a reference to the controller on the scope with the provided name.
- The audio manager has a `None` option that is useful in some UI's that need to disable sound effects by selecting it, but it's less useful for this UI. If the user adds a sound action, we want a sound to be played. Remove that option from the array returned here.
- After making those two changes, add a `sound` property to the controller (as described with `sounds`) and update the controller/template to use that instead of a property on the scope. Assign it an initial value of `this['sounds'][0]` to make sure the first sound is auto-selected when the UI launches.

Reply:
These changes are current with the last commit